File: Assets/_Project/Scripts/Data/IonSpecies.cs
using UnityEngine;

namespace Kemika.Data
{
    public enum IonCategory
    {
        Cation,
        Anion
    }

    [CreateAssetMenu(fileName = "NewIon", menuName = "Kemika/Ion Species")]
    public class IonSpecies : ScriptableObject
    {
        public string id;
        public string displayNameEn;
        public string displayNameAr;
        public IonCategory category;

        public static IonSpecies Create(string id, string nameEn, string nameAr, IonCategory category)
        {
            var ion = CreateInstance<IonSpecies>();
            ion.id = id;
            ion.displayNameEn = nameEn;
            ion.displayNameAr = nameAr;
            ion.category = category;
            return ion;
        }
    }
}

File: lib/data/ion_species.dart
enum IonCategory { cation, anion }

class IonSpecies {
  const IonSpecies({
    required this.id,
    required this.displayNameEn,
    required this.displayNameAr,
    required this.category,
  });

  final String id;
  final String displayNameEn;
  final String displayNameAr;
  final IonCategory category;
}
